Update download progress calculation in CVersionUpdater

Lpsd opened the pull request:
Resolves #2825 (?) - untested, but should get rid of any edge cases possible due to the existing code

Lpsd committed:
Update download progress calculation

Lpsd committed:
Use float

TheNormalnij commented on Client/core/CVersionUpdater.cpp:
```diff
@@ -2768,7 +2768,9 @@ int CVersionUpdater::_PollDownload()
                     if (m_JobInfo.bShowDownloadPercent)
                         GetQuestionBox().SetMessage(
-                            SString(_("%3d %% completed"), Round(m_JobInfo.uiBytesDownloaded * 100.f / std::max<float>(1, m_JobInfo.iFilesize))));
+                            SString(_("%3d %% completed"), (m_JobInfo.uiBytesDownloaded > 0 && m_JobInfo.iFilesize >= m_JobInfo.uiBytesDownloaded)
```
Human readable self documented variant
```c++
const bool bIsDownloadedSizeRight = m_JobInfo.uiBytesDownloaded > 0 && m_JobInfo.iFilesize >= m_JobInfo.uiBytesDownloaded;
const float fDownloadedPercent = bIsDownloadedSizeRight ? std::min<int>(100, Round(m_JobInfo.uiBytesDownloaded / m_JobInfo.iFilesize * 100.f)) : 0;
GetQuestionBox().SetMessage(SString(_("%3d %% completed"), fDownloadedPercent));

```

Lpsd committed:
Improve code readability

Lpsd commented:
Added the suggested changes for code readability purposes.

Lpsd committed:
Use std::max instead of std::min

We're already checking if m_JobInfo.uiBytesDownloaded is higher than 0, and if m_JobInfo.iFileSize is bigger than this, so the only possibility left is a negative number, not a value over 100

Lpsd committed:
Remove unnecessary std::max

Lpsd commented:
f7b0b1e was a brainfart, there's enough sanity checking in `bIsDownloadedSizeRight` to ensure the value produced isn't below 0 or above 100
